zigzag/zigzag_test_log.py

```diff
@@ -512,3 +536,190 @@ def _get_truncated_failure_output(cls, message):
             truncated_message += line + '\n' if len(line) <= max_line_length else line[:max_line_length] + '...\n'
 
         return truncated_message.rstrip()
+
+
+class _ZigZagTestLogWithSteps(_ZigZagTestLog):
+    def __init__(self, testcase_name, teststeps_xml, mediator):
+        """Create a TestLog object that contains steps.
+
+        Args:
+            testcase_name (str): The name of the test case.
+            teststeps_xml (list(ElementTree)): A list of JUnit style testcase XML elements representing test steps for
+                a single qTest test log.
+            mediator (ZigZag): The mediator that stores shared data.
+        """
+
+        # Initialize without parsing so we can post-process the test steps.
+        super(_ZigZagTestLogWithSteps, self).__init__(None, mediator, auto_parse=False, auto_lookup=False)
+
+        self._name = testcase_name
+        self._teststeps_xml = teststeps_xml
+        self._zz_test_step_logs = []
+        self._qtest_test_step_logs = []
+
+        self._parse()
```

Comment:
Looks like you are parsing here even though the signature says `auto_parse=False`.